File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
          long long rev=0;
       while(x!=0){
        long long digits=x%10;
        rev=rev*10+digits;
        x/=10;
       }
       if(rev>INT_MAX||rev<INT_MIN){
            return 0;
        }
       return rev;
        
    }
};

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int rev = 0;
        
        while (x != 0) {
            int n = x % 10;
            
            // Check for overflow before updating rev hello this is for the nothing
            if (rev > Integer.MAX_VALUE / 10 || (rev == Integer.MAX_VALUE / 10 && n > 7)) {
                return 0;
            }
            if (rev < Integer.MIN_VALUE / 10 || (rev == Integer.MIN_VALUE / 10 && n < -8)) {
                return 0;
            }
            
            rev = rev * 10 + n;
            x /= 10;
        }
        
        return rev;
        
    }
}
